### ISSUE TITLE
Have a PySelf ref type to improve error messages for invalid arg counts to methods

### ISSUE THREAD

**Issue by coolreader18** [RustPython/RustPython#741]
This is probably only once `arbitrary_self_types`, lands, but in CPython if you do

```
>>> l = []
>>> l.append()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: append() takes exactly one argument (0 given)
>>> list.append()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: descriptor 'append' of 'list' object needs an argument
```

The error messages are different, to indicate the different kinds of arguments that are missing (self vs normal argument). Whereas in RustPython:

```
>>>>> l = []
>>>>> l.append()
Traceback (most recent call last):
  File <stdin>, line 0, in <module>
TypeError: Expected at least 2 arguments (1 given)
>>>>> list.append()
Traceback (most recent call last):
  File <stdin>, line 0, in <module>
TypeError: Expected at least 2 arguments (0 given)
```

So you'd probably do something like

```rust
impl PyListRef {
    fn append(self: PySelf<PyList>, ...) { ... }
}
```

**Comment by youknowone** [comment]
Two more current examples of this same root cause (the receiver is counted in the arity and the callable name is missing), verified against CPython 3.14.5:

```python
>>> dict.fromkeys()
# RustPython: TypeError: expected at least 2 arguments, got 1
# CPython:    TypeError: fromkeys expected at least 1 argument, got 0

>>> int.from_bytes(b'\xff', 'big', True)   # signed is keyword-only
# RustPython: TypeError: expected at most 3 arguments, got 4
# CPython:    TypeError: from_bytes() takes at most 2 positional arguments (3 given)
```

The generic messages come from `FuncArgs::bind` / `ArgumentError::into_exception` in `crates/vm/src/function/argument.rs` (`expected at {least,most} N arguments, got M`), which reports over the raw `FromArgs` arity — so for methods/classmethods that bind `self`/`cls` through `FromArgs`, both counts are off by one and no callable name is included.

(Filed #8378 separately before finding this issue — closing that as a duplicate of this one.)

— *commented by Claude*
